src/CompleteBipartiteGraph.jl

```diff
@@ -0,0 +1,158 @@
+
+
+# =======================================================
+#         CompleteBipartiteGraph struct
+# =======================================================
+
+"""
+    CompleteBipartiteGraph <: AbstractGraph
+
+A structure representing an undirected bipartite graph.
+
+
+See also: [`LightGraphs.complete_bipartite_graph`](@ref)
+"""
+struct CompleteBipartiteGraph{T<:Integer} <: LG.AbstractGraph{T}
+    m::T
+    n::T
+
+    function CompleteBipartiteGraph{T}(m, n) where {T}
+       
+        m = convert(T, m)
+        n = convert(T, n)
+        m >= zero(T) || throw(ArgumentError("m must be >= 0"))
+        n >= zero(T) || throw(ArgumentError("n must be >= 0"))
+        _, does_overflow = Base.Checked.add_with_overflow(m, n)
+        does_overflow && throw(ArgumentError("m + n must not exceed typemax($T)"))
+
+        return new{T}(m, n)
+    end
+end
+
+CompleteBipartiteGraph(m::T, n::T) where {T<: Integer} = CompleteBipartiteGraph{T}(m, n)
+
+
+# =======================================================
+#         traits
+# =======================================================
+
+LG.is_directed(::Type{<:CompleteBipartiteGraph}) = false
+
+
+# =======================================================
+#        vertices
+# =======================================================
+
+Base.eltype(::Type{CompleteBipartiteGraph{T}}) where {T} = T
+Base.eltype(g::CompleteBipartiteGraph) = eltype(typeof(g))
+
+LG.nv(g::CompleteBipartiteGraph{T}) where {T} = T(g.m + g.n)
+
+LG.vertices(g::CompleteBipartiteGraph{T}) where {T} = Base.OneTo(nv(g))
+
+LG.has_vertex(g::CompleteBipartiteGraph, v) = v in vertices(g)
+
+
+# =======================================================
+#        edges
+# =======================================================
+
+
+LG.edgetype(::Type{CompleteBipartiteGraph{T}}) where {T} = Edge{T}
+LG.edgetype(g::CompleteBipartiteGraph{T}) where {T} = edgetype(typeof(g))
+
+function LG.ne(g::CompleteBipartiteGraph)
+    m = Int(g.m)
+    n = Int(g.n)
+
+    return m * n
+end
+
+function LG.has_edge(g::CompleteBipartiteGraph{T}, u, v) where {T}
+
+    u, v = minmax(T(u), T(v))
+
+    m = g.m
+    n = g.n
+rem
+    return (u ∈ Base.OneTo(m)) & ((v - m) ∈ (Base.OneTo(n)))
+end
+
+# ---- edges vector -----------------------------------
+
+LG.edges(g::CompleteBipartiteGraph) = SimpleEdgeVector(g)
+
+function Base.size(edgevec::SimpleEdgeVector{V, G}) where {V, G <: CompleteBipartiteGraph}
+
+    g = edgevec.graph
+
+    return (ne(g), )
+end
+
+@inline function Base.getindex(edgevec::SimpleEdgeVector{V, G}, i::Int) where {V, G <: CompleteBipartiteGraph}
+
+    @boundscheck i ∈ Base.OneTo(length(edgevec)) || throw(BoundsError(edgevec, i))
+
+    g = edgevec.graph
+    T = eltype(g)
+    m::T = g.m
```

Comment:
Isn't the type assertion always true?

Reply:
Yes, they are, but I thought it would be helpful for verifying that we calculate the correct type.

Btw. As far as I know it is not really a type assertion, when something will be assigned to `m` that is not of type `T, then Julia will try to convert it instead of failing as one would expect from an assertion.

Reply:
> Btw. As far as I know it is not really a type assertion, when something will be assigned to m that is not of type `T, then Julia will try to convert it instead of failing as one would expect from an assertion.

That's right yes I abused the word assertion